### PR TITLE
fix: add log for ipam data received

### PR DESCRIPTION
**Pull request by QxBytes**
**Reason for Change**:

Adds logs so we can tell if NMAgent response is empty/does not contain interfaces when using azure vnet ipam

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [X] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by rbtr — APPROVED** [review]

File: ipam/azure.go [hunk 1]
```diff
@@ -115,6 +115,8 @@ func (s *azureSource) refresh() error {
 		return err
 	}
 
+	log.Printf("[ipam] xml name received:%v interfaces:%d", doc.XMLName, len(doc.Interface))
```
Comment (tamilmani1989) [inline ipam/azure.go:118]: shouldn't we print whole xml output?

File: ipam/azure.go [hunk 2]
```diff
@@ -142,6 +144,8 @@ func (s *azureSource) refresh() error {
 			continue
 		}
 
+		log.Printf("[ipam] processing interface:%s IsPrimary:%t macAddress:%s", ifName, i.IsPrimary, i.MacAddress)
```
Comment (tamilmani1989) [inline ipam/azure.go:147]: log subnet too.. addresspool is created based on that

File: ipam/azure.go (hunk 1 above)
Comment (tamilmani1989) [reply ipam/azure.go:118]: ignore.. that will be lot if user has configured 200+ Ipconfigs..

File: ipam/azure.go [hunk 3]
```diff
@@ -156,6 +168,7 @@ func (s *azureSource) refresh() error {
 				continue
 			}
 
+			log.Printf("[ipam] all ipaddresses found in ipsubnet struct: %d", len(s.IPAddress))
```
Comment (tamilmani1989) [inline ipam/azure.go:171]:
```suggestion
			log.Printf("[ipam] Number of IPAddress found in ipsubnet %s: %d", len(s.IPAddress), s.Prefix)
```

File: ipam/azure.go [hunk 4]
```diff
@@ -115,6 +118,13 @@ func (s *azureSource) refresh() error {
 		return err
 	}
 
+	log.Printf("[ipam] xml name received:%v interfaces:%d", doc.XMLName, len(doc.Interface))
+
+	if len(doc.Interface) == 0 {
```
Comment (tamilmani1989) [inline ipam/azure.go:123]: can we log doc in this case?

Comment (QxBytes) [reply ipam/azure.go:123]: if we log doc won't it log the entire struct with all the ip configs?

**Review by tamilmani1989 — APPROVED** [review]

**Pull request merged**